src/mlpack/methods/ann/modelParser.cpp

```diff
@@ -0,0 +1,1132 @@
+
+  map<string, double> initDetails;
+  string initType;
+  BOOST_FOREACH (ptree::value_type const &v, init.get_child(""))
+  {
+    const ptree &attributes = v.second;
+    if (v.first == "type")
+    {
+      initType = attributes.get_value<string>();
+    }
+    else
+    {
+      initDetails[v.first] = attributes.get_value<double>();
+    }
+  }
+  cout << "Init details:\ntype : " << initType << endl;
+  printMap(initDetails);
+  
+  map<string, double> optimizerDetails;
+  string optimizerType;
+  BOOST_FOREACH (ptree::value_type const &v, optimizer.get_child(""))
+  {
+    const ptree &attributes = v.second;
+    if (v.first == "type")
+    {
+      optimizerType = attributes.get_value<string>();
+    }
+    else
+    {
+      optimizerDetails[v.first] = attributes.get_value<double>();
+    }
+  }
+  cout << "\nOptimizer details:\ntype : " << optimizerType << endl;
+  printMap(optimizerDetails);
+  cout << "\nNetwork details:\n\n";
+  BOOST_FOREACH (ptree::value_type const &v, network.get_child(""))
+  {
+    const ptree &layerWhole = v.second;
+    map<string, double> params;
+    string layerType;
+    BOOST_FOREACH (ptree::value_type const &v2, layerWhole.get_child(""))
+    {
+      const ptree &layerInner = v2.second;
+      //cout << v2.first << "\t";
+      //cout << layerInner.get_value<string>() << endl;
+      string key = boost::erase_all_copy(v2.first, "_");
+      boost::to_lower(key);
+      if (key == "type")
+      {
+        layerType = layerInner.get_value<string>();
+      }
+      else if (key == "units")
+      {
+        params["insize"] = inSize;
+        inSize = params["outsize"] = layerInner.get_value<double>();
+      }
+      else
+      {
+        if(key == "outsize")
+          inSize = layerInner.get_value<double>();
+        params[key] = layerInner.get_value<double>();
+      }
+    }
+    cout << "type : " << layerType << endl;
+    printMap(params);
+    layers.push(getNetworkReference(layerType, params));
+    cout << endl;
+  }
+  getInitType(initType, lossType, initDetails, optimizerType,
+      optimizerDetails, layers, dataset);
+}
+
+/**
+ * Create a propert tree from the given json file
+ * 
+ * @param fileName Path to the json file from which the network
+ * properties would be loaded
+ * @param dataset The Dataset object that contains the training and validation
+ * data
+ * @param inSize The input size of the first layer
+ */
+boost::property_tree::ptree loadProperties(string fileName, Dataset& dataset,
+                                           double inSize)
+{
+  ptree pt;
+  read_json(fileName, pt);
+  traverseModel(pt, dataset, inSize);
+  return pt;
+}
+
+/** The final implementation of this file would not have a main method. This is 
+ * merely to ease testing. The following include satement can hence be 
+ * removed later
+*/
+#include <mlpack/core/data/split_data.hpp>
+int main()
+{
+  //arma::mat trainX, trainY;
+  string fileName = "network3.json";
```

Comment:
Perhaps you can post this file somewhere? Is the json file autogenerated?

Reply:
Sorry for the delay. I had completely missed this comment. 
The json file is user specified. I think it will be a lot clearer if you take a look [here](https://github.com/sreenikSS/Testing/blob/master/Dataset3.json)